docs/reference/esql/functions/search.asciidoc

```diff
@@ -0,0 +1,14 @@
+[discrete]
+[[esql-search-operators]]
+=== Search operators
+
+The only search operator is match (`:`):
+
+preview::["Do not use on production environments. This functionality is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features."]
+
+[.text-center]
+image::esql/functions/signature/match_operator.svg[Embedded,opts=inline]
+
+Supported types:
+
+include::types/match.asciidoc[]
```

Comment:
I think for doing that it would be needed to add a new `MatchOperator` class with the corresponding `@FunctionInfo` annotation that contains the examples, so they are auto-generated and can be linked.

We could create that class just for doc purposes and avoid instantiating it, or make it subclass `Match` - it could even be used to represent the operator instead of using `Match` for both the function and the operator.

WDYT?

Reply:
The `search.asciidoc` is not a generated file, we can just add an example of match operator manually without adding a new class. `in.asciidoc` has an example of including a csvtest there.

Reply:
Nice! I didn't know we could add CSV tests directly without generating the examples via tests. Thanks!

Reply:
Added an example, now search operators looks like:

![image](https://github.com/user-attachments/assets/2be2dbb2-f3b4-4956-bf38-54b7a60b39c8)